lumicks/pylake/detail/widefield.py

```diff
@@ -10,6 +10,16 @@
 from ..adjustments import ColorAdjustment
 
 
+def _get_page_timestamp(page, start=True):
+    """Get the starting timestamp from a TIFF page"""
+    regex = r"^(\d+):\d+$" if start else r"^\d+:(\d+)$"
+    timestamp_string = re.search(regex, page.tags["DateTime"].value)
+    if timestamp_string:
+        return np.int64(timestamp_string.group(1))
+    else:
+        raise ValueError("Incorrectly formatted timestamp metadata")
```

Comment:
Should we maybe just get both timestamps here rather than checking if it's start or stop? Doesn't seem to me like it'd be a performance hit and the `start=bool` seems a bit odd to me. This is essentially the logic you want for `frame_timestamp_ranges` anyway

Reply:
Good point. I'll add an extra commit that fixes this up.